### PR TITLE
Replace getCurrentException and getCurrentExceptionMsg

### PR DESCRIPTION
**Pull request by kdeme**
Similar as for the other repos, change to stop using `getCurrentException`. From going over them, didn't actually notice any that could create an issue.

### CODE REVIEW

File: eth/net/nat.nim [hunk 1]
```diff
@@ -71,8 +71,8 @@ proc getExternalIP*(natStrategy: NatStrategy, quiet = false): Option[IpAddress]
             externalIP = parseIpAddress(ires.value)
             strategy = NatUpnp
             return some(externalIP)
-          except:
-            error "parseIpAddress() exception", err = getCurrentExceptionMsg()
+          except ValueError as e:
```
Comment (kdeme) [inline eth/net/nat.nim]: For reference, this is the only error that can be raised currently.
Could change it though to `CatchableError` if we don't trust this not to change as we have no raises pragma to guard it.

File: eth/p2p/discovery.nim [hunk 2]
```diff
@@ -263,10 +263,12 @@ proc processClient(transp: DatagramTransport,
...
     proto.receive(a, buf)
-  except RlpError, DiscProtocolError:
-    debug "Receive failed", err = getCurrentExceptionMsg()
-  except:
-    debug "Receive failed", err = getCurrentExceptionMsg()
+  except RlpError as e:
+    debug "Receive failed", exc = e.name, err = e.msg
+  except DiscProtocolError as e:
+    debug "Receive failed", exc = e.name, err = e.msg
+  except Exception as e:
```
Comment (stefantalpalaru) [inline eth/p2p/discovery.nim]: Don't you want CatchableError here?

File: eth/p2p/peer_pool.nim [hunk 3]
```diff
@@ -173,8 +173,7 @@ proc run(p: PeerPool) {.async.} =
       # Most unexpected errors should be transient, so we log and restart from
```
Comment (stefantalpalaru) [inline eth/p2p/peer_pool.nim]: The exception above should probably be CatchableError.

File: eth/p2p/rlpx.nim [hunk 4]
```diff
@@ -443,11 +443,11 @@ proc checkedRlpRead(peer: Peer, r: var Rlp, MsgType: type): auto {.inline.} =
   else:
     try:
       return r.read(MsgType)
-    except:
+    except Exception as e:
```
Comment (stefantalpalaru) [inline eth/p2p/rlpx.nim]: Guess what?

File: eth/p2p/discovery.nim (hunk 2 above)
Comment (kdeme) [reply eth/p2p/discovery.nim]: It's being reraised, so no, not necessarily. I guess it is open for debate (see libp2p fix discussion).
Anyway, In this case it makes sense to just remove it.

File: eth/p2p/peer_pool.nim (hunk 3 above)
Comment (kdeme) [reply eth/p2p/peer_pool.nim]: True, I'll change it.

File: eth/p2p/rlpx.nim (hunk 4 above)
Comment (kdeme) [reply eth/p2p/rlpx.nim]: Same story, here it is used to add some extra useful logging before the application dies. Is this only useful for CatchableErrors and not for Defects? I'm not so sure, I remember hitting quite some AssertionsErrors and other Defects when fuzzing rlpx and rlp.

**Pull request merged**